docs/register-create-stake-phase-2.md

```diff
@@ -0,0 +1,401 @@
+  // staking_value  is the amount of satoshis locked in staking output
+  int64 staking_value = 6;
+  // staking_tx is a bitcoin staking transaction i.e transaction that locks funds
+  bytes staking_tx = 7 ;
+  // staking_tx_inclusion_proof is the inclusion proof of the staking tx in BTC chain
+  InclusionProof staking_tx_inclusion_proof = 8;
+  // slashing_tx is the slashing tx
+  // Note that the tx itself does not contain signatures, which are off-chain.
+  bytes slashing_tx = 9 [ (gogoproto.customtype) = "BTCSlashingTx" ];
+  // delegator_slashing_sig is the signature on the slashing tx by the delegator (i.e., SK corresponding to btc_pk).
+  // It will be a part of the witness for the staking tx output.
+  // The staking tx output further needs signatures from covenant and finality provider in
+  // order to be spendable.
+  bytes delegator_slashing_sig = 10 [ (gogoproto.customtype) = "github.com/babylonlabs-io/babylon/types.BIP340Signature" ];
+  // unbonding_time is the time lock used when funds are being unbonded. It is be used in:
+  // - unbonding transaction, time lock spending path
+  // - staking slashing transaction, change output
+  // - unbonding slashing transaction, change output
+  // It must be smaller than math.MaxUInt16 and larger that max(MinUnbondingTime, CheckpointFinalizationTimeout)
+  uint32 unbonding_time = 11;
+  // fields related to unbonding transaction
+  // unbonding_tx is a bitcoin unbonding transaction i.e transaction that spends
+  // staking output and sends it to the unbonding output
+  bytes unbonding_tx = 12;
+  // unbonding_value is amount of satoshis locked in unbonding output.
+  // NOTE: staking_value and unbonding_value could be different because of the difference between the fee for staking tx and that for unbonding
+  int64 unbonding_value = 13;
+  // unbonding_slashing_tx is the slashing tx which slash unbonding contract
+  // Note that the tx itself does not contain signatures, which are off-chain.
+  bytes unbonding_slashing_tx = 14 [ (gogoproto.customtype) = "BTCSlashingTx" ];
+  // delegator_unbonding_slashing_sig is the signature on the slashing tx by the delegator (i.e., SK corresponding to btc_pk).
+  bytes delegator_unbonding_slashing_sig = 15 [ (gogoproto.customtype) = "github.com/babylonlabs-io/babylon/types.BIP340Signature" ];
+}
+```
+
+We will go through the fields in the above message in detail to understand how 
+to construct them.
+
+### Staking Transaction
+
+In the staker program, a staking transaction is created as a Bitcoin transaction 
+that locks funds into a specific staking output. This transaction is 
+identified by its hash.
+
+For the `MsgCreateBTCDelegation` message, you must include the staking 
+transaction hash. This hash is used by the Babylon chain to verify 
+the transaction and ensure the funds are correctly locked for staking.
+
+To generate this hash you will need to do the following:
+
+1. Build the staking hash using `BuildStakingInfo`:
```

Comment:
The following are a bit hard to follow, not sure why need them here. We don't need to provide code examples, that's not the goal of this document. It should serve as a "router" document which explains to user the sequence of steps of what they need to do and provides them links to the places they can find details if they don't know them.